package/harvester-os/files/usr/sbin/gen-kdump-initrd

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash -ex
+
+KERNEL_VERSION=$(uname -r)
+INITRD_KDUMP_NAME="initrd-$KERNEL_VERSION-kdump"
+INITRD_CACHE_DIR="/usr/local/cache/kdump"
+
+if [ -e /boot/$INITRD_KDUMP_NAME ]; then
+  echo /boot/$INITRD_KDUMP_NAME exists. Skip generating a new one.
+  exit 0
+fi
+
+if [ -e $INITRD_CACHE_DIR/$INITRD_KDUMP_NAME ]; then
+  echo "Using cache: $INITRD_CACHE_DIR/$INITRD_KDUMP_NAME"
+  cp $INITRD_CACHE_DIR/$INITRD_KDUMP_NAME /boot/$INITRD_KDUMP_NAME
+  exit 0
+fi
+
+mkdir -p $INITRD_CACHE_DIR
+
+TMP_DIR=$(mktemp -d)
+trap "rm -rf $TMP_DIR" exit
+
+TMP_INITRD=$TMP_DIR/initrd
+INITRD_EXTRACTED_DIR=$TMP_DIR/extracted
+DRACUT_CONFDIR=$TMP_DIR/dracut.conf.d
+
+mkdir -p $INITRD_EXTRACTED_DIR
+mkdir -p $DRACUT_CONFDIR
+
+
+# specify a new dracut config dir to avoid including cOS configs in /etc/dracut.conf.d
+cat > $DRACUT_CONFDIR/99-tmp.conf <<EOF
+early_microcode="no"
+host_only="yes"
+EOF
+/usr/bin/dracut --confdir $DRACUT_CONFDIR --force --omit "plymouth resume usrmount" --add kdump $TMP_INITRD $KERNEL_VERSION
+
+# extract initrd and replace fstab and kdump config
+cd $INITRD_EXTRACTED_DIR
+xzcat $TMP_INITRD | cpio -idm
+
+PERSISTENT_PARTITION=$(lsblk -n -o UUID /dev/disk/by-label/COS_PERSISTENT)
+cat > etc/fstab <<EOF
+UUID=$PERSISTENT_PARTITION /kdump/mnt0 auto defaults 0 0
+EOF
```

Comment:
NIT: maybe this works?

```suggestion
cat > etc/fstab <<EOF
/dev/disk/by-label/COS_PERSISTENT /kdump/mnt0 auto defaults 0 0
EOF
```

Just not sure about the usage of this mount and the modification of `KDUMP_SAVEDIR`. Could you explain them more? Thanks.

Reply:
That's my original version.
But I found in some case user have multiple partitions with the `COS_PERSISTENT` label (e.g. another disk was installed before).
I need to make sure the dump is saving to the current OS's persistent partition.